src/deterministic_brb.rs

```diff
@@ -367,12 +408,30 @@ impl<A: Actor<S>, SA: SigningActor<A, S>, S: Sig, BRBDT: BRBDataType<A>>
                     .or_default()
                     .push((msg.clone(), proof));
 
-                // Remove the message from pending_proof since we now have proof
-                self.pending_proof.remove(&msg);
-
                 // Apply the op
-                self.dt.apply(msg.op);
+                self.dt.apply(msg.op.clone());
 
+                Ok(vec![self.send(
+                    msg.dot.actor,
+                    Payload::BRB(Op::Delivered { msg }),
+                )?])
+            }
+            Op::Delivered { msg } => {
+                let number_of_confirms = if let Some(confirms) = self.pending_delivery.get_mut(&msg)
+                {
+                    confirms.insert(source);
+                    confirms.len()
+                } else {
+                    0
+                };
+
+                if self.supermajority(number_of_confirms, msg.gen)? {
+                    // We've seen a super-majority of delivery confirmations so we can
+                    // be confident this operation has been committed.
+                    self.pending_delivery.remove(&msg);
+                    // Remove the message from pending_proof since we have now received delivery confirmation.
+                    self.pending_proof.remove(&msg);
```

Comment:
Hmm, a couple of points re moving pending_proof.remove() inside here.

1. Doesn't this change the meaning of pending proof?   ie, in definition comment it is defined as:  "Msgs this process has initiated and is waiting on BFT agreement for from the network".   But now that seems to extend to: I received agreement + delivery confirmations.    That may be fine, but perhaps the comment could be made more explicit.

2. Doesn't this make pending_proof and pending_delivery mean essentially the same thing?    ie, could we not just use pending_proof and get rid of pending_delivery, for purposes of checking if a proof has been delivered to supermajority or not?

Reply:
I was re-using the pending_proof for sending out any undelivered ProofOfAgreements since pending_proof was storing the signatures needed to build up the ProofOfAgreement packet.

I've made a change to untangle these two by copying the proof into the `pending_delivery` buffer to so that we can clear out pending proof independently of `pending_delivery`